pkg/provider/cloud/kubevirt/data_volume.go

```diff
@@ -32,28 +32,93 @@ import (
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	preAllocatedDVAnnotation  = "cdi.kubevirt.io/storage.deleteAfterCompletion"
+	preAllocatedDVSize        = "11Gi"
+	osAnnotationForCustomDisk = "cdi.kubevirt.io/os-type"
+)
+
 func dataVolumeCreator(datavolume *cdiv1beta1.DataVolume) reconciling.NamedCDIv1beta1DataVolumeCreatorGetter {
 	return func() (name string, create reconciling.CDIv1beta1DataVolumeCreator) {
 		return datavolume.Name, func(dv *cdiv1beta1.DataVolume) (*cdiv1beta1.DataVolume, error) {
+			dv.Annotations = datavolume.Annotations
 			dv.Spec = datavolume.Spec
 			return dv, nil
 		}
 	}
 }
 
-func reconcilePreAllocatedDataVolumes(ctx context.Context, cluster *kubermaticv1.Cluster, client ctrlruntimeclient.Client) error {
+func reconcileDataVolume(ctx context.Context, client ctrlruntimeclient.Client, dataVolume *cdiv1beta1.DataVolume, namespace string) error {
+	dvCreator := []reconciling.NamedCDIv1beta1DataVolumeCreatorGetter{
+		dataVolumeCreator(dataVolume),
+	}
+	return reconciling.ReconcileCDIv1beta1DataVolumes(ctx, dvCreator, namespace, client)
+}
+
+// reconcileLocalDataVolumes reconciles the local-disks from cluster.
+func reconcileLocalDataVolumes(ctx context.Context, cluster *kubermaticv1.Cluster, client ctrlruntimeclient.Client) error {
 	for _, d := range cluster.Spec.Cloud.Kubevirt.PreAllocatedDataVolumes {
 		dv, err := createPreAllocatedDataVolume(d, cluster.Status.NamespaceName)
 		if err != nil {
 			return err
```

Comment:
If one of the erroneous local-disk is used for the creation of MD, then it leads to more errors.
So just **fail fast** in case of error and return.

Reply:
but if we continue, we would not reconcile the erroneous dv right ?
We would just reconcile the one that do not have errors (with the current code, it would be an error on the size).